tests/get_servers_test.rs

```diff
@@ -0,0 +1,75 @@
+#[cfg(test)]
+mod tests {
+    use std::env;
+    use std::net::{Ipv4Addr, SocketAddr};
+    use std::time::Duration;
+
+    // These would normally be imported from your crate
+    // use your_crate::get_servers;
+    // use your_crate::ip_range;
+
+    // Simple test for get_servers CIDR environment parsing
+    // This test doesn't make actual network requests
+    #[tokio::test]
+    async fn test_get_servers_cidr_parsing() {
+        // Set environment variable for testing
+        env::set_var("NETWORK_CIDR", "192.168.1.0/30");
+
+        // We need to directly test the CIDR parsing logic from get_servers()
+        // For this, we'll extract the relevant parts to test
+
+        let cidr = env::var("NETWORK_CIDR").unwrap();
+        let network = ipnetwork::Ipv4Network::from_str(&cidr).expect("Invalid CIDR format");
+
+        // Assert that the network is parsed correctly
+        assert_eq!(network.network(), Ipv4Addr::new(192, 168, 1, 0));
+        assert_eq!(network.broadcast(), Ipv4Addr::new(192, 168, 1, 3));
+
+        // Clean up
+        env::remove_var("NETWORK_CIDR");
+    }
```

Comment:
If we [switch to clap](https://docs.rs/clap/latest/clap/) instead of the half assed ENV VAR parsing we can externalize the parsing of the string to CIDNCIDR.